tests/service/data/TestTP/ProvidedTypes.fsi

```diff
@@ -119,6 +119,12 @@ type ProvidedMethod =
     /// Define the static parameters available on a statically parameterized method
     member DefineStaticParameters: parameters: ProvidedStaticParameter list * instantiationFunction: (string -> obj[] -> ProvidedMethod) -> unit
 
+    member DefineStaticParameters: parameters: ProvidedStaticParameter list * instantiationFunction: (ProvidedMethod -> string -> obj[] -> ProvidedMethod) -> unit
+
+    member ReportWarning: staticParameterName: string * rangeInParameterIfString: (int * int) option * message: string * throwIfNotSupported: bool -> unit
+
+    member ReportError: staticParameterName: string * rangeInParameterIfString: (int * int) option * message: string -> unit
+
```

Comment:
I'm really not happy with adding these 3 methods (and with the reference to self in `instantiationFunction` in particular), but I could not think of a way to report diagnostics for static method arguments by calling `ReportWarning/Error` through the provider type itself, as is the case when applying type-level static arguments.